presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -71,18 +71,24 @@
     private final ScheduledExecutorService configExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("DbResourceGroupConfigurationManager"));
     private final AtomicBoolean started = new AtomicBoolean();
     private final String environment;
+    private final boolean enableExactMatchSelector;
 
     @Inject
-    public DbResourceGroupConfigurationManager(ClusterMemoryPoolManager memoryPoolManager, ResourceGroupsDao dao, @ForEnvironment String environment)
+    public DbResourceGroupConfigurationManager(ClusterMemoryPoolManager memoryPoolManager, DbResourceGroupConfig config, ResourceGroupsDao dao, @ForEnvironment String environment)
     {
         super(memoryPoolManager);
         requireNonNull(memoryPoolManager, "memoryPoolManager is null");
+        requireNonNull(config, "config is null");
+        this.enableExactMatchSelector = config.getEnableExactMatchSelector();
```

Comment:
nit: please move this line below `requireNonNull(dao, "daoProvider is null");` to keep all non-null checks together and at the beginning of the method.

Reply:
I thought we check non-null right before using, no?

Reply:
interesting, in many different parts of the code I got the impression of bundling these checks together.